app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SignedOutController.scala

```diff
@@ -30,16 +32,26 @@ import scala.concurrent.Future
 
 @Singleton
 class SignedOutController @Inject()(
-  knownFactsResultMongoRepository: KnownFactsResultMongoRepository,
+  chainedSessionRepository: ChainedSessionDetailsRepository,
+  mappingConnector: MappingConnector,
   sessionStoreService: SessionStoreService)(implicit appConfig: AppConfig)
     extends FrontendController {
 
   def redirectToSos = Action.async { implicit request =>
     for {
       knownFactOpt <- sessionStoreService.fetchKnownFactsResult
       id <- knownFactOpt match {
-             case Some(knownFact) => knownFactsResultMongoRepository.create(knownFact).map(Option.apply)
-             case None            => Future successful None
+             case Some(knownFact) => {
+               for {
+                 isEligibleForMapping <- mappingConnector.isEligibile
+                 _ <- if (isEligibleForMapping) mappingConnector.createPreSubscription(knownFact.utr)
+                     else Future.successful(())
+                 idOpt <- chainedSessionRepository
+                           .create(ChainedSessionDetails(knownFact, isEligibleForMapping))
+                           .map(Option.apply)
+               } yield idOpt
```

Comment:
this ID should never be an option. Is this possible?

Reply:
Nice, thanks